lib/handlers/method_not_allowed.ex

```diff
@@ -0,0 +1,22 @@
+defmodule HTTPServer.Handlers.MethodNotAllowed do
```

Comment:
`lib/handlers` now contains 4 `HTTPServer.Handlers` that are "built in", shared, and with very little potential to change.  I wonder if they should be separated from any kind of user build handlers?

For example, since they are all exclusively used by the `HTTPServer.Router` I wonder if they should be considered part of it?  `lib/router/handlers` and `HTTPServer.Router.Handlers` (or something like that)?

Reply:
I actually went back and forth on whether to create that previous [Campsite PR for the Options Hander](https://github.com/kelseyroy/http_server/pull/17) because I did feel like handlers wasn't exactly the right place for it. I knew I wanted that logic out of the router file, but I didn't really have any other place to put it. I like this suggestion of creating a new file and module category for handlers that are fixed and therefore should not be associated with client handlers!

Reply:
Question: since Redirect is called and handled in the routes themselves (not in Router) should it remain in the handlers directory while Options, MethodNotAllowed and NotFound get moved to `HTTPServer.Router.Handlers`?

Reply:
The `Router` and the `routes.ex` files are coupled in a way I think makes sense to move them all.

Reply:
You may want to change that design in the future, but right now its not needed and no one is asking for it.

Reply:
Redirect, Options, MethodNotAllowed and NotFound now exist in `lib/router/handlers`. I left router alone but could move it into `lib/router` if you think it needs to be included in the directory as well.